lib/std/Build.zig

```diff
@@ -1525,6 +1525,17 @@ pub fn step(b: *Build, name: []const u8, description: []const u8) *Step {
     return &step_info.step;
 }
 
+pub fn standardLinkageOption(b: *Build, target: Target) std.builtin.LinkMode {
+    if (b.option(std.builtin.LinkMode, "linkage", "Method of linking binaries")) |linkage| {
+        return linkage;
+    }
+
+    return switch (target.os.tag) {
+        .linux, .macos, .ios, .windows => .dynamic,
+        else => .static,
+    };
```

Comment:
This is definitely going to be controversial, FWIW. I know Andrew strongly prefers static linking by default, when possible.

Reply:
Gotcha, yeah I was just thinking it's more of ABI restricted rather than architecture or OS restricted.

Reply:
Ok, I've updated it now so it's going to use dynamic linking when it's actually required.